### ISSUE TITLE
Minor issue in auto.gum()

### ISSUE THREAD

**Issue by Steviey** [config-i1/greybox#49]
Hi!

I get a code break when trying to fit a model with auto.gum and type='select'.
The system then tries to fit a multiplicative model and fails with a code break.
I saw this behavior before, from package forecast, trying seasonal where there is no seasonality.

The problem is, I'm running multiple fits with various inputs and attributes in a loop.
It would be some how convenient, if it would switch to additive automatically, if it can't fit a multiplicative model.

I can hint this with an explicit exception... so not really a bug... 

Checking model with a type="a".
Starting preliminary loop: 1 out of 1. Done.
Searching for appropriate lags: We found them!
Searching for appropriate orders: Orders found.
Checking model with a type="m".
Starting preliminary loop:           Error in costfunc(matvt, matF, matw, yInSample, vecg, h, lagsModel, Etype,  : 
  Mat::operator(): index out of bounds
     █
  1. ├─global::bAlgos(i, f, "prediction") ~/R/daScript.R:51786:16
  2. │ └─global::standAloneAlgos(...) ~/R/daScript.R:51399:16
  3. │   └─smooth::auto.gum(...) ~/R/daScript.R:23035:16
  4. │     └─smooth::gum(...)
  5. │       └─smooth:::CreatorGUM(silentText = silentText)
  6. │         └─nloptr::nloptr(...)
  7. ├─(function (x) ...
  8. │ └─smooth:::eval_f(x, ...)
  9. │   └─smooth:::costfunc(...)
 10. ├─base::stop(...)
 11. └─(function () ...
 12.   └─lobstr::cst() ~/R/daScript.R:9:29
No traceback available

**Comment by config-i1** [comment]
Hi,

1. This is not a greybox function, but smooth one.
2. Can you, please, open issue there (https://github.com/config-i1/smooth/issues/) and provide a reproducible example?

Cheers,
Ivan

**Comment by Steviey** [comment]
Ok, but I don't think it's a real bug/issue.

**Comment by config-i1** [comment]
Well, anyway, I can look into it, as long as I can reproduce it :)

**Comment by Steviey** [comment]
It should have been related to the input data, because I interchanged it, with success in the one case and fail in the other. I also took a test with positive data only and it failed. But anyways, could be, that I made an unknown error. Meanwhile an exception for failed models is in place. Thank you for your help Ivan. By the way, there is something related to the error message here: https://github.com/metrumresearchgroup/mrgsolve/issues/399 
...but I saw no reason to dig deeper.
